### PR TITLE
Added service to set camera's follow offset

### PR DESCRIPTION
**Pull request by jennuine**
# 🎉 New feature

## Summary
Added service `/gui/follow/offset` to set the user camera's follow offset.

## Test it
```
ign gazebo diff_drive.sdf

ign service -s /gui/follow --reqtype ignition.msgs.StringMsg --reptype ignition.msgs.Boolean --timeout 2000 --req 'data: "vehicle_blue"'

# sets camera to face vechile_blue
ign service -s /gui/follow/offset --reqtype ignition.msgs.Vector3d --reptype ignition.msgs.Boolean --timeout 2000 --req "x: 5, y: 0, z: 3"
```

## Checklist
- [X] Signed all commits for DCO
- [ ] Added tests
- [ ] Added example and/or tutorial
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [x] `codecheck` passed (See [contributing](https://ignitionrobotics.org/docs/all/contributing#contributing-code))
- [ ] All tests passed (See [test coverage](https://ignitionrobotics.org/docs/all/contributing#test-coverage))
- [X] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Aignitionrobotics+repo%3Aosrf%2Fsdformat+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge**

### CODE REVIEW

**Review by iche033 — APPROVED** [review]
nice, works for me

**Comment by iche033** [comment]
while review this PR, I found a small bug related to the camera follow offset checks and addressed it in https://github.com/ignitionrobotics/ign-gazebo/pull/857

**Comment by nkoenig** [comment]
I'll try this out with some log files soon. Please wait for my approval before merging.

**Review by nkoenig — APPROVED** [review]

**Pull request merged**
